packages/components/utils/src/getStringMatch/getStringMatch.mdx

```diff
@@ -0,0 +1,81 @@
+---
+title: 'getStringMatch'
+type: 'component'
+slug: /utils/get-string-match/
+typescript: ./getStringMatch.ts
+v4Doc: true
+---
+
+import { Props } from '@contentful/f36-docs-utils';
+
+Function that can be used to find a substring inside another string.
+
+## Table of contents
+
+- [Import](#import)
+- [Examples](#examples)
+  - [Basic usage](#basic-usage)
+  - [Example with other components](#example-with-other-components)
+- [Props (API reference)](#props-api-reference)
+
+## Import
+
+```jsx static=true
+import { getStringMatch } from '@contentful/f36-utils';
+```
+
+## Examples
+
+### Basic usage
+
+You pass the base string as a first argument, and a string to match it with as a second argument. The function will return an object containing the match, what comes before the match, and what comes after the match in the base string.
+
+```jsx
+() => {
+  const [baseString, setBaseString] = React.useState('Seven Green Apples');
+  const [stringToMatch, setStringToMatch] = React.useState('green');
+  const [{ before, match, after }, updateResult] = React.useState(() =>
+    getStringMatch(baseString, stringToMatch),
+  );
+
+  const matchString = () => {
+    const { before, match, after } = getStringMatch(baseString, stringToMatch);
+    updateResult({ before, match, after });
+  };
+
+  return (
+    <Stack flexDirection="column">
+      <Flex>
+        <FormControl>
+          <FormControl.Label>Base string</FormControl.Label>
+
+          <TextInput
+            value={baseString}
+            onChange={(e) => setBaseString(e.target.value)}
+          />
+        </FormControl>
+        <FormControl>
+          <FormControl.Label>
+            A string to match the base string
+          </FormControl.Label>
+          <TextInput
+            value={stringToMatch}
+            onChange={(e) => setStringToMatch(e.target.value)}
+          />
+        </FormControl>
+      </Flex>
+      <Button onClick={matchString}>Match strings</Button>
+      <Heading as="h4">Results</Heading>
+      <Paragraph element="p">{`{ before: "${before}", match: "${match}", after: "${after}" }`}</Paragraph>
+    </Stack>
+  );
+};
+```
+
+### Example with other components
+
+You can find an example of using `getStringMatch` in [`Autocomplete` component documentation](/components/autocomplete#highlighting-an-item-with-getstringmatch)
+
+## Props (API reference)
+
+<Props of="getStringMatch" />
```

Comment:
![Screenshot 2021-12-07 at 15 06 22](https://user-images.githubusercontent.com/6597467/145044445-edee35f9-979c-4e6f-85b4-dafdf8e43e6c.png)

It looks like `react-docgen-typescript` is not generating the props of this function correctly, but I think this bug we should fix in another PR/story

I will write a ticket for that